code/src/main/kotlin/com/expediagroup/sdk/lodgingconnectivity/configuration/ClientConfiguration.kt

```diff
@@ -16,147 +16,179 @@
-         * @param requestTimeout The request timeout duration.
-         */
-        fun requestTimeout(requestTimeout: Long) = apply {
-            this.requestTimeout = requestTimeout
-        }
-
-        /**
-         * Sets the connection timeout in milliseconds.
-         * @param connectionTimeout The connection timeout duration.
-         */
-        fun connectionTimeout(connectionTimeout: Long) = apply {
-            this.connectionTimeout = connectionTimeout
-        }
-
-        /**
-         * Sets the socket timeout in milliseconds.
-         * @param socketTimeout The socket timeout duration.
-         */
-        fun socketTimeout(socketTimeout: Long) = apply {
-            this.socketTimeout = socketTimeout
-        }
-
-        /**
-         * Sets the headers whose values should be masked in logs.
-         * @param maskedLoggingHeaders A set of HTTP headers to mask in logs.
-         */
-        fun maskedLoggingHeaders(maskedLoggingHeaders: Set<String>) = apply {
-            this.maskedLoggingHeaders = maskedLoggingHeaders
-        }
-
-        /**
-         * Sets the body fields whose values should be masked in logs.
-         * @param maskedLoggingBodyFields A set of fields in the request body to mask in logs.
-         */
-        fun maskedLoggingBodyFields(maskedLoggingBodyFields: Set<String>) = apply {
-            this.maskedLoggingBodyFields = maskedLoggingBodyFields
-        }
-
-        /**
-         * Builds and returns the `ClientConfiguration` instance.
-         * @return The configured `ClientConfiguration`.
-         */
-        fun build(): ClientConfiguration {
-            return ClientConfiguration(
-                key,
-                secret,
-                environment,
-                requestTimeout,
-                connectionTimeout,
-                socketTimeout,
-                maskedLoggingHeaders,
-                maskedLoggingBodyFields
+        fun interceptors(interceptors: List<Interceptor>) = apply {
+            this.interceptors = interceptors
+        }
+
+        fun networkInterceptors(networkInterceptors: List<Interceptor>) = apply {
+            this.networkInterceptors = networkInterceptors
+        }
+
+        fun connectionPool(connectionPool: ConnectionPool) = apply {
+            this.connectionPool = connectionPool
+        }
+
+        fun retryOnConnectionFailure(retryOnConnectionFailure: Boolean) = apply {
+            this.retryOnConnectionFailure = retryOnConnectionFailure
+        }
+
+        fun callTimeout(callTimeout: Int) = apply {
+            this.callTimeout = callTimeout
+        }
+
+        fun connectTimeout(connectTimeout: Int) = apply {
+            this.connectTimeout = connectTimeout
+        }
+
+        fun readTimeout(readTimeout: Int) = apply {
+            this.readTimeout = readTimeout
+        }
+
+        fun writeTimeout(writeTimeout: Int) = apply {
+            this.writeTimeout = writeTimeout
+        }
```

Comment:
Why do the 2 required fields return `this`, but none of the other builder methods? Shouldn't they?

Reply:
changed it to use `apply` as the others 👍🏻

Reply:
Any other builders we should update?

Reply:
Updated the `Headers` builder, I don't find any others